validation/helpers.py

```diff
@@ -88,7 +88,7 @@ def diff_char_or_none(index):
 
         # end check for swapping glides
 
-        if c.split()[1] in vowels:
+        if len(c.split()) > 1 and c.split()[1] in vowels:
```

Comment:
It's not obvious to me what exactly this conditional is testing for. A refactoring that would make this more readable would be Extract Function/[Decompose Conditional](https://refactoring.com/catalog/decomposeConditional.html):

so:

```suggestion
        if has_vowel_condition_idk_jolene_plz_rename(c):
```

```python
def has_vowel_condition_idk_jolene_plz_rename(c):
   components = c.split()
   return len(components) > 1 and component[1] in vowels
```

Reply:
Most tokens come with a '-' or a '+' in front of them, but sometimes they just have a space. So while most split items will have a length of 2, some will make it to this point and only have a length of 1, meaning it wasn't added or removed from the original word to create the suggested word, so the code can just carry on. I can extract it into a function, though, if that will make it clearer.